testutil/integration/model_test.go

```diff
@@ -11,105 +29,205 @@ import (
-	"github.com/kleveross/klever-model-registry/pkg/registry/models"
 )
 
 var _ = Describe("Model Registry", func() {
 	const timeout = time.Second * 5
 	const interval = time.Second * 1
 
+	var modelExtract = []struct {
+		model   string
+		version string
+		format  modeljobsv1alpha1.Format
+	}{
+
+		{"onnx", "v1", modeljobsv1alpha1.FormatONNX},
+		{"savedmodel", "v1", modeljobsv1alpha1.FormatSavedModel},
+		{"graphdef", "v1", modeljobsv1alpha1.FormatGraphDef},
+		{"netdef", "v1", modeljobsv1alpha1.FormatNetDef},
+		{"h5", "v1", modeljobsv1alpha1.FormatH5},
+		{"caffemodel", "v1", modeljobsv1alpha1.FormatCaffeModel},
+		{"mxnetparams", "v1", modeljobsv1alpha1.FormatMXNETParams},
+		{"torchscript", "v1", modeljobsv1alpha1.FormatTorchScript},
+		{"pmml", "v1", modeljobsv1alpha1.FormatPMML},
+		//{"tensorrt","v1", "TensorRT"},
+	}
+
+	var modelConvert = []struct {
+		model   string
+		tomodel string
+		version string
+		format  modeljobsv1alpha1.Format
+		toformat modeljobsv1alpha1.Format
+
+	}{
+		{"h5", "tensorflow","v1", modeljobsv1alpha1.FormatH5, modeljobsv1alpha1.FormatSavedModel},
+		{"caffemodel", "netdef","v1", modeljobsv1alpha1.FormatCaffeModel, modeljobsv1alpha1.FormatNetDef},
+		{"mxnetparams", "onnx", "v1", modeljobsv1alpha1.FormatMXNETParams, modeljobsv1alpha1.FormatONNX},
+	}
+
 	e := httpexpect.New(GinkgoT(), ModelRegistryHost)
 
 	Context("Models", func() {
 		project := "library"
-		model := "tensorflow"
-		version := "test"
-
-		It("Should push the model successfully", func() {
-			modelContent := models.Model{
-				ProjectName: project,
-				ModelName:   model,
-				VersionName: version,
-				Format:      "SavedModel",
+		//check the model status
+		It("Should list the model successfully", func() {
+			//Upload the model by ormb
+			var tool ormb.Interface
+			var plainHTTPOpt bool
+			var insecureOpt bool
+			project := "library"
+			rootPath, err := filepath.Abs(viper.GetString("./models"))
+			if err != nil{
+				panic(err)
 			}
-			bytes, err := json.Marshal(modelContent)
-			Expect(err).Should(BeNil())
-
-			e.POST("/api/v1alpha1/projects/{projectName}/models/{modelName}/versions/{versionName}/upload",
-				project, model, version).
-				WithHeaders(map[string]string{
-					"X-Tenant": "test",
-					"X-User":   "test",
-				}).
-				WithMultipart().
-				WithFile("file", "./models/model.zip").WithFormField("model", string(bytes)).
-				Expect().Status(http.StatusCreated)
-			// Upload model will create ModelJob automatically, now get ModelJobList.
-			Eventually(func() error {
-				length := e.GET("/api/v1alpha1/namespaces/{namespace}/modeljobs/",
-					"default").Expect().JSON().Object().Value("items").Array().Length()
-				if int(length.Raw()) != 1 {
-					return fmt.Errorf("Not found any modeljob")
-				}
-				return nil
-			}, timeout, interval).Should(Succeed())
+			logrus.WithFields(logrus.Fields{
+				"root-path": rootPath,
+			}).Debugln("Create the ormb client with the given root path")
 
+			tool, err = ormb.New(
```

Comment:
all of the err should be check, eg:
Expect(err).Should(BeNil())

Reply:
OK~